types/git-username/index.d.ts

```diff
@@ -5,8 +5,10 @@
 
 export = gitUsername;
 
+declare function gitUsername(options?: gitUsername.BaseOptions): string | null;
 declare function gitUsername(options: gitUsername.StrictOptions): string;
-declare function gitUsername(options?: string | gitUsername.BaseOptions): string | null;
+declare function gitUsername(cwd: string, options?: gitUsername.BaseOptions): string | null;
+declare function gitUsername(cwd: string, options?: gitUsername.StrictOptions): string;
```

Comment:
I think we can get rid of all of this `string | null` debacle. This function can always return a `null` independent of the value of the `strict` option. So this file should actually look something like this:

```ts
declare function gitUsername(cwd?: string, options?: gitUsername.BasicOptions): string | null;
declare function gitUsername(options: gitUsername.FullOptions): string | null;

declare namespace gitUsername {
    interface BasicOptions {
        strict?: boolean;
    }

    interface FullOptions extends BasicOptions {
        cwd?: string;
    }
}
```